### DISCUSSION TITLE
Functionality for "variable nullability" of columns

### DISCUSSION THREAD

**Discussion by petegriffith** [prisma/prisma#16644] [Q&A]
Does Prisma have functionality to allow one of two columns to be nullable, but not both? And also not neither?

**Comment by nurul3101** [comment]
Hello @petegriffith 👋 

I don't think you can specify this validation in Prisma Schema, but you can create a migration with --create-only flag - [Reference](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#customizing-migrations) and add a CHECK constraint which validates these null conditions.

Here is an [example](https://stackoverflow.com/questions/63061242/how-to-make-sure-only-one-column-is-not-null-in-postgresql-table) of how you can enforce a null condition check constraint.

**Reply by vladi-stevanovic** [reply]
👋 Hi @petegriffith , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
